### PR TITLE
Task-58643: Fix rename document uploaded

### PR DESCRIPTION
**Pull request by sofyenne**
ISSUE: When user rename document uploaded the extension of document not displayed 
FIX : change the clean string method to correctly clean the new name

### CODE REVIEW

File: documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/util/JCRDocumentsUtil.java [hunk 1]
```diff
@@ -534,7 +534,7 @@ public static String cleanString(String str) {
     int i = 0;
     while (i < strLength) {
       char c = cleanedStr.charAt(i);
-      if (c == '/' || c == ':' || c == '[' || c == ']' || c == '*' || c == '\'' || c == '"' || c == '|' || c == 'ʿ' || c == 'ˇ') {
+      if (c == '/' || c == ':' || c == '[' || c == ']' || c == '*' || c == '\'' || c == '"' || c == '|' || c == 'ʿ' || c == 'ˇ' || c == '.') {
         cleanedStr.deleteCharAt(i);
         cleanedStr.insert(i, '_');
       } else if (!(Character.isLetterOrDigit(c) || Character.isWhitespace(c) || c == '.' || c == '-' || c == '_')) {
```
Comment (ahamdi) [inline documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/util/JCRDocumentsUtil.java]: the dot character is already removed here

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/util/JCRDocumentsUtil.java (hunk 1 above)
Comment (sofyenne) [reply documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/util/JCRDocumentsUtil.java]: done !! i change the else statement , when rename document the string to clean is the title of document without the extension so we need to replace the dot character by "_" not remove it to keep the meaning of the name  .
example : document name = `test.test.test.pdf` => expected name is    `test_test_test.pdf `

File: documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java [hunk 2]
```diff
@@ -536,7 +536,7 @@ public void renameDocument(long ownerId, String documentID, String title, Identi
       String oldName = node.getName();
       if (oldName.indexOf('.') != -1 && node.isNodeType(NodeTypeConstants.NT_FILE)) {
         String ext = oldName.substring(oldName.lastIndexOf('.'));
-        title = title.concat(ext);
+        title = name.concat(ext);
```
Comment (ahamdi) [inline documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java:538]: No need to change the title, the title is a property that does not need to be cleaned nor escaped its illegal JCR characters

**Review by ahamdi — CHANGES_REQUESTED** [review]
Please add a UT if possible

File: documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java (hunk 2 above)
Comment (ahamdi) [reply documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java:538]: Also I don't get why do we need to do cleanString twice in lines 530 and 532

Comment (sofyenne) [reply documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java:538]: i remove the unnecessary clean string

**Review by ahamdi — APPROVED** [review]

**Pull request merged**